x-pack/legacy/plugins/maps/public/layers/sources/es_search_source/es_search_source.js

```diff
@@ -279,20 +292,31 @@ export class ESSearchSource extends AbstractESSource {
       };
     }
 
+    const cardinalityAgg = { precision_threshold: 1 };
+    const termsAgg = {
+      size: DEFAULT_MAX_BUCKETS_LIMIT,
+      shard_size: DEFAULT_MAX_BUCKETS_LIMIT,
+    };
+    const topHitsSplitField = this._getField(indexPattern, topHitsSplitFieldName);
+    if (topHitsSplitField.scripted) {
```

Comment:
do you think it;'s worth creating a utility out of this? Something like

```
function addFieldToDSL(dsl, field) {
   if (!field.scripted) {
      dsl.field = field;
     return;
   }

   dsl.script = {
    source: field.script,
    lang: field.lang
  };
}
```

and reuse:

- https://github.com/elastic/kibana/blob/455a96e107b72bf147700cfdc155a8f129ba0310/x-pack/legacy/plugins/maps/public/layers/fields/es_doc_field.js#L68
- https://github.com/elastic/kibana/blob/455a96e107b72bf147700cfdc155a8f129ba0310/x-pack/legacy/plugins/maps/public/layers/fields/es_doc_field.js#L45